src/main/java/uk/gov/pay/ledger/transaction/service/TransactionService.java

```diff
@@ -151,6 +158,44 @@ private TransactionSearchResponse searchTransactionsAndParent(TransactionSearchP
         return transactionListForCsv;
     }
 
+    public List<TransactionEntity> searchTransactionAfter(TransactionSearchParams searchParams) {
+        return transactionDao.cursorTransactionSearch(searchParams, null, null);
+    }
+
+    public List<TransactionEntity> searchTransactionAfter(TransactionSearchParams searchParams, ZonedDateTime startingAfterCreatedDate, Long startingAfterId) {
+        return transactionDao.cursorTransactionSearch(searchParams, startingAfterCreatedDate, startingAfterId);
+    }
+
+    public Map<String, Object> csvHeaderFrom(TransactionEntity sample) {
+        return csvTransactionFactory.getSimpleCsvHeaders();
+    }
+
+    public String csvStringFrom(Map<String, Object> headers) throws JsonProcessingException {
+        return csvString(List.of(headers), headers);
+    }
+
+    public String csvStringFrom(List<TransactionEntity> page, Map<String, Object> headers) throws JsonProcessingException {
+        List<Map<String, Object>> csvTransactions = page
+                .stream()
+                .map(csvTransactionFactory::toMap)
+                .collect(Collectors.toList());
+
+        return csvString(csvTransactions, headers);
+    }
+
+    private String csvString(List<Map<String, Object>> page, Map<String, Object> headers) throws JsonProcessingException {
+        CsvMapper mapper = new CsvMapper();
```

Comment:
This is expensive to create for each transaction. Current logic in CSVMessageBodyWrite could still work except that the transactions list is queried from DB in batches
https://github.com/alphagov/pay-ledger/blob/ae835b70fc10023fa6d901bd5802b2f0d3687c31/src/main/java/uk/gov/pay/ledger/util/csv/CSVMessageBodyWriter.java#L43

Reply:
Really good catch! @kbottla reference this performance wiki https://github.com/FasterXML/jackson-docs/wiki/Presentation:-Jackson-Performance#basics-things-you-should-do-anyway